### PR TITLE
Hardcoded README.md (#31)

### PR DESCRIPTION
**Pull request by bolshoytoster**
This update makes the README.md path dynamic using the `SOFT_SERVE_README_PATH` environment variable.

### CODE REVIEW

**Comment by bolshoytoster** [comment]
I fixed the build error

**Comment by toby** [comment]
Thank you for this! It's really cool (and a good idea). Just a couple of notes... Can you change the `ReadmePath` arg in the function definitions to `rmp` to match the style of the rest of the args? Not the cfg.ReadmePath, that's fine. Also, can you squash this PR into a single commit?

Other than that, it looks and works great!

File: internal/git/git.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ func (rs *RepoSource) GetCommits(limit int) []RepoCommit {
 }
 
 // LoadRepos opens Git repositories.
-func (rs *RepoSource) LoadRepos() error {
+func (rs *RepoSource) LoadRepos(ReadmePath string) error {
```
Comment (toby) [inline internal/git/git.go:120]: Change to `rmp` here

File: internal/git/git.go [hunk 2]
```diff
@@ -141,7 +141,7 @@ func (rs *RepoSource) LoadRepos() error {
 	return nil
 }
 
-func (rs *RepoSource) loadRepo(name string, rg *git.Repository) (*Repo, error) {
+func (rs *RepoSource) loadRepo(name string, rg *git.Repository, ReadmePath string) (*Repo, error) {
```
Comment (toby) [inline internal/git/git.go:144]: and `rmp` here as well

**Comment by bolshoytoster** [comment]
@toby alright, I only named it `ReadMePath` because it’s referred as that in other parts of the repo.

File: config/config.go [hunk 3]
```diff
@@ -35,6 +36,9 @@ func (c *Config) applyDefaults() {
 	if c.RepoPath == "" {
 		c.RepoPath = ".repos"
 	}
+  if c.ReadmePath == "" {
+    c.ReadmPath = "README.md"
```
Comment (toby) [inline config/config.go:40]: Hi @bolshoytoster this looks like a typo that's causing it to fail to build. Should be `c.ReadmePath` not `c.ReadmPath`.

Comment (toby) [reply config/config.go]: No worries! I think you may have rebased a bit too far back as this includes some earlier changes and skips a commit. Can you do a fetch and rebase your changes on top of 2a1313d? There's also a missing `_`, I'll leave a comment in PR where it's at.

File: README.md [hunk 4]
```diff
@@ -160,6 +160,7 @@ environment-level settings:
 * `SOFT_SERVE_HOST`: SSH listen host (_default 0.0.0.0_)
 * `SOFT_SERVE_KEY_PATH`: SSH host key-pair path (_default .ssh/soft_serve_server_ed25519_)
 * `SOFT_SERVE_REPO_PATH`: Path where repos are stored (_default .repos_)
+* `SOFT_SERVE_README_PATH`: Path where the README.md is created (_default README.md)
```
Comment (toby) [inline README.md:163]: This should read `(_default README.md_)`. It's just missing the last _.

**Comment by brunnre8** [comment]
Not to throw a wrench into this, but does it make sense as a global?

After all where to put the readme (and its name) is repo specific and depends on the project/user.

Wouldn't it be better to make this configurable per repo?

Just to give you an idea, here's a pretty standard cgit config:
```
readme=:README.md
readme=:readme.md
readme=:README.mkd
readme=:readme.mkd
readme=:README.rst
readme=:readme.rst
readme=:README.html
readme=:readme.html
readme=:README.htm
readme=:readme.htm
readme=:README.txt
readme=:readme.txt
readme=:README
readme=:readme
readme=:INSTALL.md
readme=:install.md
readme=:INSTALL.mkd
readme=:install.mkd
readme=:INSTALL.rst
readme=:install.rst
readme=:INSTALL.html
readme=:install.html
readme=:INSTALL.htm
readme=:install.htm
readme=:INSTALL.txt
readme=:install.txt
readme=:INSTALL
readme=:install
```

**Comment by bolshoytoster** [comment]
@brunnre8
I could add it to this config:

https://github.com/charmbracelet/soft-serve/blob/e8b51469d2d72d7e86a648333553c4230f76b200/README.md#L62-L111

```yaml
repos: 
   - name: Lorem 
     repo: ipsum 
     private: false 
     note: "dolor sit amet"
     readme: consectetur/README.md
```

**Comment by aymanbagabas** [comment]
You can now specify a readme file path per repository in your `config.yaml`, otherwise, it finds the first match of `README*` and uses for the readme. Example `readme: docs/README.md`
https://github.com/charmbracelet/soft-serve/pull/98

Closing...

**Pull request closed without merging**